contrib/inventory/foreman.py

```diff
@@ -0,0 +1,394 @@
+        except (ConfigParser.NoOptionError, ConfigParser.NoSectionError):
+            self.want_facts = True
+
+        # Cache related
+        try:
+            cache_path = os.path.expanduser(config.get('cache', 'path'))
+        except (ConfigParser.NoOptionError, ConfigParser.NoSectionError):
+            cache_path = '.'
+        (script, ext) = os.path.splitext(os.path.basename(__file__))
+        self.cache_path_cache = cache_path + "/%s.cache" % script
+        self.cache_path_inventory = cache_path + "/%s.index" % script
+        self.cache_path_params = cache_path + "/%s.params" % script
+        self.cache_path_facts = cache_path + "/%s.facts" % script
+        try:
+            self.cache_max_age = config.getint('cache', 'max_age')
+        except (ConfigParser.NoOptionError, ConfigParser.NoSectionError):
+            self.cache_max_age = 60
+        return True
+
+    def parse_cli_args(self):
+        """Command line argument processing"""
+
+        parser = argparse.ArgumentParser(description='Produce an Ansible Inventory file based on foreman')
+        parser.add_argument('--list', action='store_true', default=True, help='List instances (default: True)')
+        parser.add_argument('--host', action='store', help='Get all the variables about a specific instance')
+        parser.add_argument('--refresh-cache', action='store_true', default=False,
+                            help='Force refresh of cache by making API requests to foreman (default: False - use cache files)')
+        self.args = parser.parse_args()
+
+    def _get_json(self, url, ignore_errors=None):
+        page = 1
+        results = []
+        while True:
+            ret = requests.get(url,
+                               auth=HTTPBasicAuth(self.foreman_user, self.foreman_pw),
+                               verify=self.foreman_ssl_verify,
+                               params={'page': page, 'per_page': 250})
+            if ignore_errors and ret.status_code in ignore_errors:
+                break
+            ret.raise_for_status()
+            json = ret.json()
+            # /hosts/:id has not results key
+            if 'results' not in json:
+                return json
+            # Facts are returned as dict in results not list
+            if isinstance(json['results'], dict):
+                return json['results']
+            # List of all hosts is returned paginaged
+            results = results + json['results']
+            if len(results) >= json['total']:
+                break
+            page += 1
+            if len(json['results']) == 0:
+                print("Did not make any progress during loop. "
+                      "expected %d got %d" % (json['total'], len(results)),
+                      file=sys.stderr)
+                break
+        return results
+
+    def _get_hosts(self):
+        return self._get_json("%s/api/v2/hosts" % self.foreman_url)
+
+    def _get_hostgroup_by_id(self, hid):
```

Comment:
Yes, it was used to resolve Foreman host parameters on https://github.com/theforeman/foreman_ansible_inventory/commit/9972f3fff005a065ea6ebaee73e9fc66d0cf8961#diff-2b3c7ba3ec283a3b380cff70f61e0290R144 , not needed anymore